cmd/benchttp/run.go

```diff
@@ -82,21 +78,21 @@ func (cmd *cmdRun) execute(args []string) error {
 // parseArgs parses input args as config fields and returns
 // a slice of fields that were set by the user.
 func (cmd *cmdRun) parseArgs(args []string) []string {
-	// config file path: always read it so it can be
-	// properly defaulted if not set.
-	cmd.flagset.StringVar(&cmd.configFile,
-		"configFile",
-		configfile.Find(cmd.defaultConfigFiles),
-		"Config file path",
-	)
-
 	// first arg is subcommand "run"
 	// skip parsing if no flags are provided
 	if len(args) <= 1 {
 		return []string{}
 	}
 
+	// config file path
+	cmd.flagset.StringVar(&cmd.configFile,
+		"configFile",
+		cmd.configFile,
```

Comment:
I'm curious: is there a way to defer the evaluation of the default file until we are sure the flag was not provided ?
 Because we can totally search for the default file via `configfile.Find` and then override it with cli flag. Making a file system search for nothing.

I know this was already the behavior as `configfile.Find(cmd.defaultConfigFiles)` as the second argument to `flagset.StringVar` would evaluate immediatly, but seeing the refactor makes me think about this.

Reply:
Interesting! We can achieve this easily by reversing the logic: first set `cmd.configFile` to the value of `-configFile` (with a default value of `""`) then if `cmd.configFile == ""` we fallback to the default files.

I'll give a try to that approach but I might prefer the current one if the new one turns out to be more complex or less straightforward.

Reply:
Should we merge as it is and change it in another PR if we have the time to?
I agree that ideally, not searching for the file if we override it would be better, but it does not seem critical to me.